index.js

```diff
@@ -1,136 +1,27 @@
 'use strict';
-var fs = require('fs');
-var util = require('util');
-var tmp = require('tmp');
-
-var DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
-
-function rethrow() {
-  // Only enable in debug mode. A backtrace uses ~1000 bytes of heap space and is fairly slow to generate.
-  if (DEBUG) {
-    var backtrace = new Error();
-    return function(err) {
-      if (err) {
-        backtrace.stack = err.name + ': ' + err.message +
-          backtrace.stack.substr(backtrace.name.length);
-        err = backtrace;
-        throw err;
-      }
-    };
-  }
-
-  return function(err) {
-    if (err) {
-      throw err; // Forgot a callback but don't know where? Use NODE_DEBUG=fs
-    }
-  };
-}
-
-function maybeCallback(callback) {
-  return typeof callback === 'function' ? callback : rethrow();
-}
-
-module.exports = function prependFile(path, data, options) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
-
-  if (typeof options === 'function' || !options) {
-    options = {
-      encoding: 'utf8',
-      mode: 438 /*=0666*/
-    };
-  } else if (util.isString(options)) {
-    options = {
-      encoding: options,
-      mode: 438
-    };
-  } else if (!util.isObject(options)) {
-    throw new TypeError('Bad arguments');
+const fs = require('fs');
+const {promisify} = require('util');
+const tempWrite = require('temp-write');
+const pathExists = require('path-exists');
+const pipeline = promisify(require('stream').pipeline);
+
+module.exports = async (filename, data) => {
+  if (await pathExists(filename)) {
+    const temporaryFile = await tempWrite(data);
```

Comment:
Hi @Richienb, why is it necessary to write to a temp file first?

Reply:
https://stackoverflow.com/a/15424018/8384910

Reply:
Hmm I see. I think I have an optimization in mind though. Might open a PR in the future.